packages/esm-reports-app/src/components/run-report/run-report-form.component.tsx

```diff
@@ -54,22 +54,19 @@ const RunReportForm: React.FC<RunReportForm> = ({ closePanel }) => {
     switch (parameter.type) {
```

Comment:
A few things here:

1. I think it makes most sense to hoist this function outside of the `RunReportForm` component. Because of how React hooks work, that's likely to be better for performance.
2. I think we can give better typings to `parameter`, i.e., it seems like its something like this:

```ts
interface ReportParameter {
  name: string;
  type: string;
  label: string;
}
```

3. Keys aren't really useful or necessary here.

Reply:
> Keys aren't really useful or necessary here.
I did this because I was trying to resolve an issue with the parameters not being cleared when a new report was selected, so I did this to ensure the components get reinitialised every time a new report is generated.  There is probably a better way of achieving the same behaviour.